### PR TITLE
docs: document loading related modules

### PR DESCRIPTION
**Pull request by kopp**
## Description

`py::module_` is not known to my `pybind11` but `py::module` is.  Also I've added a use-case which was not mentioned in the documentation before.

### CODE REVIEW

**Comment by YannickJadoul** [comment]
> `py::module_` is not known to my `pybind11` but `py::module` is.

Then you need to update pybind11 (or read the old documentation). `py::module_` is now recommended (just like e.g. `class_` or `enum_`), to avoid clashes with C++20's modules.

**Comment by YannickJadoul** [comment]
Feel free to further describe your issue, btw.

> Also I've added a use-case which was not mentioned in the documentation before.

Please reopen if you still want to get this in? Also, you'll need to target `master` instead of `stable`, normally.

**Comment by kopp** [comment]
Ah, I'm sorry. I was assuming that my system uses the latest release which obviously it does not. 

If you think that this usecase is an actual use (and not a misuse) I'll reopen a separate PR for it. 

The master/stable thing is because I clicked the "edit in GitHub" link from readthedocs

**Comment by YannickJadoul** [comment]
No problem!

> If you think that this usecase is an actual use (and not a misuse) I'll reopen a separate PR for it.

Sure; or you can repurpose this one, if you want.

> The master/stable thing is because I clicked the "edit in GitHub" link from readthedocs

Right, but as you see, CI is missing some fixes on `stable`, so if necessary to release a patch, we'll backport anything from `master` to `stable` ;-)

**Comment by kopp** [comment]
Hi @YannickJadoul ,  it looks like the CI is happy now :-)  Feel free to merge the PR if you think that it adds value to the documentation.

**Review by YannickJadoul — APPROVED** [review]

**Review by rwgk — CHANGES_REQUESTED** [review]
@kopp did you see my questions?
General: IMO the added section lacks precision/rigor. It needs more careful wording.
Specific:
1. how is "related native" defined? It's unclear to me. People will have to guess (not good). If you are referring to "another pybind11 extension", I'd say "related native" is highly misleading (certainly to people not familiar with pybind11 already).
2. the example error message hides that the reported type information is for C++ types and can easily make people think they could be Python types. Again, this can send people down the wrong path. It should be clear that the type information is for C++ types, and that there is no firm connection to Python module names (pretty likely), or that even the Python type names could be different (probably unusual). That doesn't need to be explained in detail, but should be made clear by picking a good example and more careful wording.

File: docs/advanced/pycpp/object.rst [hunk 1]
```diff
@@ -65,8 +65,28 @@ This example obtains a reference to the Python ``Decimal`` class.
     return scipy.attr("__version__");
 
 
+.. _loading_related_native_modules:
```
[the remark this answers is not in the document]
Comment (YannickJadoul) [reply docs/advanced/pycpp/object.rst:68]: > What is "related native"? From the context it looks like you mean "another pybind11 extension", is that correct?

"related [native modules]", not "[related native] modules"

**Comment by YannickJadoul** [comment]
> @kopp did you see my questions?

You must have not submitted these before, because I did also not see any activity until 15 minutes ago on this PR.

**Comment by EricCousineau-TRI** [comment]
> [...] because I did also not see any activity on this PR until 15 minutes ago.

Perhaps via Gitter?

**Pull request closed without merging**